### PR TITLE
shm 0.30: fix some safety issues and cleanup

### PR DESCRIPTION
**Pull request by i509VCB**

### CODE REVIEW

File: src/wayland/shm/pool.rs
Comment (i509VCB) [inline src/wayland/shm/pool.rs:191]: context here is that these functions are now unsafe.

**Review by vberger — APPROVED** [review]
Maybe we should merge this once the `DataInit` stuff is fixed in wayland-rs (I have a good idea what needs to be done) before merging this, so that we don't forget the related `FIXME` ?

**Comment by i509VCB** [comment]
@vberger I will write a seperate pull request to update wayland-rs to the new alpha so the fixme is resolved. I have removed the comment for now under the assumption the update is done.

**Pull request merged**
